### PR TITLE
watch: support build with ncurses

### PR DESCRIPTION
**Pull request by zchee**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

watch: support build with ncurses.
Basically, avoid not supported(in other words, not included Apple's terminfo) terminfo such as following error.

```console
Error opening terminal: xterm-direct.
```

```console
$ otool -L $(brew --prefix watch)/bin/watch
/usr/local/opt/watch/bin/watch:
        /usr/local/opt/ncurses/lib/libncursesw.6.dylib (compatibility version 6.0.0, current version 6.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.250.1)
```

### CODE REVIEW

**Comment by fxcoudert** [comment]
If the bug is only present with non-default terminals, we might not want to add the extra dependency, which affects everyone.

**Comment by zchee** [comment]
@fxcoudert `non-default` means non Apple's built-in ncurses terminfo?
I tried `xterm-direct` which included ncurses formula, not exists Apple's ncurses. Yeah, `xterm-direct` is very new terminfo. It's a not bug.

But AFAIK, Apple's built-in ncurses is older. many terminfo are not include such as `screen-256color-bce-s` (IIRC)
Still doesn't use the newer version ncurses?

**Comment by fxcoudert** [comment]
No, we're going to stick to Apple's curses unless there is functionality that is broken for all users, i.e. with "standard" terminal settings.
